test/policy_integration/dnspolicy_controller_test.go

```diff
@@ -381,1115 +355,1162 @@ var _ = Describe("DNSPolicy", Ordered, func() {
+										})),
+										PointTo(MatchFields(IgnoreExtras, Fields{
+											"DNSName":          Equal("lb-" + lbHash + ".test.example.com"),
+											"Targets":          ConsistOf("default.lb-" + lbHash + ".test.example.com"),
+											"RecordType":       Equal("CNAME"),
+											"SetIdentifier":    Equal("default"),
+											"RecordTTL":        Equal(v1alpha1.TTL(300)),
+											"ProviderSpecific": Equal(v1alpha1.ProviderSpecific{{Name: "geo-code", Value: "*"}}),
+										})),
+										PointTo(MatchFields(IgnoreExtras, Fields{
+											"DNSName":       Equal(TestHostOne),
+											"Targets":       ConsistOf("lb-" + lbHash + ".test.example.com"),
+											"RecordType":    Equal("CNAME"),
+											"SetIdentifier": Equal(""),
+											"RecordTTL":     Equal(v1alpha1.TTL(300)),
+										})),
+									),
+								}),
+							}),
+							MatchFields(IgnoreExtras, Fields{
+								"ObjectMeta": HaveField("Name", wildcardRecordName),
+								"Spec": MatchFields(IgnoreExtras, Fields{
+									"ManagedZoneRef": HaveField("Name", "mz-example-com"),
+									"Endpoints": ContainElements(
+										PointTo(MatchFields(IgnoreExtras, Fields{
+											"DNSName":       Equal("19sc9b.lb-" + lbHash + ".example.com"),
+											"Targets":       ConsistOf(TestIPAddressOne, TestIPAddressTwo),
+											"RecordType":    Equal("A"),
+											"SetIdentifier": Equal(""),
+											"RecordTTL":     Equal(v1alpha1.TTL(60)),
+										})),
+										PointTo(MatchFields(IgnoreExtras, Fields{
+											"DNSName":          Equal("default.lb-" + lbHash + ".example.com"),
+											"Targets":          ConsistOf("19sc9b.lb-" + lbHash + ".example.com"),
+											"RecordType":       Equal("CNAME"),
+											"SetIdentifier":    Equal("19sc9b.lb-" + lbHash + ".example.com"),
+											"RecordTTL":        Equal(v1alpha1.TTL(60)),
+											"ProviderSpecific": Equal(v1alpha1.ProviderSpecific{{Name: "weight", Value: "120"}}),
+										})),
+										PointTo(MatchFields(IgnoreExtras, Fields{
+											"DNSName":          Equal("lb-" + lbHash + ".example.com"),
+											"Targets":          ConsistOf("default.lb-" + lbHash + ".example.com"),
+											"RecordType":       Equal("CNAME"),
+											"SetIdentifier":    Equal("default"),
+											"RecordTTL":        Equal(v1alpha1.TTL(300)),
+											"ProviderSpecific": Equal(v1alpha1.ProviderSpecific{{Name: "geo-code", Value: "*"}}),
+										})),
+										PointTo(MatchFields(IgnoreExtras, Fields{
+											"DNSName":       Equal(TestHostWildcard),
+											"Targets":       ConsistOf("lb-" + lbHash + ".example.com"),
+											"RecordType":    Equal("CNAME"),
+											"SetIdentifier": Equal(""),
+											"RecordTTL":     Equal(v1alpha1.TTL(300)),
+										})),
+									),
+								}),
+							}),
+						))
+				}, TestTimeoutMedium, TestRetryIntervalMedium, ctx).Should(Succeed())
+			})
+
```

Comment:
This is a pretty great test, but there is a lot going on in this file now. Perhaps as a follow up we could think about whether we should split this test in someway to make it easier to read / mantain? IE single vs multi or something

Reply:
yes, its far too long now, and a lot more could be added so we need to look at how we want these to be separated out. Probably at the very least a separate test file for multi/single and simple/loadbalanced contexts.